### PR TITLE
[Resources.AWS] Rename AWS detector namespace

### PR DESCRIPTION
**Pull request by joegoldman2**
Contributes to #1610.

## Changes

- Renames OpenTelemetry.ResourceDetectors.AWS to OpenTelemetry.Resources.AWS
- Makes detectors internal
- Adds `AddAWSEBSDetector`, `AddAWSEC2Detector`, `AddAWSECSDetector` and `AddAWSEKSDetector` extension methods for ResourceBuilder

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: src/OpenTelemetry.Resources.AWS/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,15 @@
   ([#1576](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1576))
 * Update OpenTelemetry SDK version to `1.8.1`.
   ([#1668](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1668))
+* **Breaking Change**: Renamed package from `OpenTelemetry.ResourceDetectors.AWS`
+  to `OpenTelemetry.Resources.AWS`.
+  ([#1820](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1839))
```
Comment (matt-hensley) [inline src/OpenTelemetry.Resources.AWS/CHANGELOG.md:14]: Link text has incorrect PR ID

Comment (joegoldman2) [reply src/OpenTelemetry.Resources.AWS/CHANGELOG.md]: Good catch!

**Review by cijothomas — APPROVED** [review]
Thanks! 
Changelog entries for breaking change advisory is very clear.

**Comment by joegoldman2** [comment]
@Kielek I'll take care of the merge conflicts.

File: src/OpenTelemetry.Resources.AWS/README.md [hunk 2]
```diff
@@ -1,20 +1,20 @@
 # AWS Resource Detectors
 
-[![NuGet version badge](https://img.shields.io/nuget/v/OpenTelemetry.ResourceDetectors.AWS)](https://www.nuget.org/packages/OpenTelemetry.ResourceDetectors.AWS)
-[![NuGet download count badge](https://img.shields.io/nuget/dt/OpenTelemetry.ResourceDetectors.AWS)](https://www.nuget.org/packages/OpenTelemetry.ResourceDetectors.AWS)
+[![NuGet version badge](https://img.shields.io/nuget/v/OpenTelemetry.Resources.AWS)](https://www.nuget.org/packages/OpenTelemetry.Resources.AWS)
+[![NuGet download count badge](https://img.shields.io/nuget/dt/OpenTelemetry.Resources.AWS)](https://www.nuget.org/packages/OpenTelemetry.Resources.AWS)
 [![codecov.io](https://codecov.io/gh/open-telemetry/opentelemetry-dotnet-contrib/branch/main/graphs/badge.svg?flag=unittests-ResourceDetectors.AWS)](https://app.codecov.io/gh/open-telemetry/opentelemetry-dotnet-contrib?flags[0]=unittests-ResourceDetectors.AWS)
```
Comment (Kielek) [inline src/OpenTelemetry.Resources.AWS/README.md:5]:
```suggestion
[![codecov.io](https://codecov.io/gh/open-telemetry/opentelemetry-dotnet-contrib/branch/main/graphs/badge.svg?flag=unittests-ResourceDetectors.AWS)](https://app.codecov.io/gh/open-telemetry/opentelemetry-dotnet-contrib?flags[0]=unittests-Resources.AWS)
```

**Review by Kielek — COMMENTED** [review]
@joegoldman2, could you please resolve conflicts after merging similar PR for Azure?

One small fix to make the codecov badge working after renames.

File: src/OpenTelemetry.Resources.AWS/README.md (hunk 2 above)
Comment (joegoldman2) [reply src/OpenTelemetry.Resources.AWS/README.md]: Good catch, thank you for your review. Fixed.

**Review by Kielek — APPROVED** [review]
LGTM, @ppittle could you please review it?

**Review by ppittle — APPROVED** [review]

**Pull request merged**